test/integration/isolated/ctx.go

```diff
@@ -0,0 +1,120 @@
+//go:build integration_tests
+
+package isolated
+
+import (
+	"context"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+)
+
+type CtxKey[R any] string
+
+func getCtxKey[R any](t *testing.T) CtxKey[R] {
+	t.Helper()
+
+	// When we pass t.Name() from inside an `assess` step, the name is in the form TestName/Features/Assess.
+	if strings.Contains(t.Name(), "/") {
+		return CtxKey[R](strings.Split(t.Name(), "/")[0])
+	}
+
+	// When we pass t.Name() from inside a `testenv.BeforeEachTest` function, the name is just TestName.
+	return CtxKey[R](t.Name())
+}
+
+func SetInCtxForT[R any](ctx context.Context, t *testing.T, r R) context.Context {
+	t.Helper()
+
+	return context.WithValue(ctx, getCtxKey[R](t), r)
+}
+
+func GetFromCtxForT[R any](ctx context.Context, t *testing.T) R {
+	t.Helper()
+
+	raw := ctx.Value(getCtxKey[R](t))
+	result, ok := raw.(R)
+	if !ok {
+		var r R
+		t.Fatalf("required %T to be stored in context but found: %s (of type %T)", r, raw, raw)
+	}
+	return result
+}
+
+func setInCtx[KeyT comparable, R any](ctx context.Context, key KeyT, r R) context.Context {
+	return context.WithValue(ctx, key, r)
+}
+
+// -----
+// RunID
+// -----
+
+type _runID struct{}
+
+func SetRunIDInCtx(ctx context.Context, runID string) context.Context {
```

Comment:
I've added godocs to those `func`s but the types I've left as is since those are not meant to be used by users. They are just to distinguish the stored/retrieved values from the context.

6137ee4fe

Reply:
👍 To clarify: what I meant by "with an explanation of what the particular value means" was to mention in the public functions' godocs what particular values (`runID`, `udpURL`) mean as a concept as for the reader their meaning might be not obvious.

Reply:
This is already explained e.g. for the run ID case, where it's created: https://github.com/Kong/kubernetes-ingress-controller/blob/1c2eb68f8c6b7f0879d1a80f262e21d049b185d3/test/integration/isolated/suite_test.go#L53-L55